src/core/query.ts

```diff
@@ -588,6 +606,16 @@ export class Query<
           isPaused: false,
           status: 'success',
         }
+      case 'ignore':
+        return {
+          ...state,
+          error: null,
+          fetchFailureCount: 0,
+          isFetching: false,
+          isInvalidated: false,
+          isPaused: false,
+          status: typeof state.data === 'undefined' ? 'idle' : 'success',
+        }
```

Comment:
why do we need this please? Can we just _not_ dispatch anything in case of bail-out and let the query be in that state as if `setQueryData` was never called?

Reply:
If a query function returns undefined or the same data, we need to reset state after bailing out. Otherwise the query will stay in fetching state.